include/swift/SwiftRemoteMirror/SwiftRemoteMirror.h

```diff
@@ -188,7 +188,7 @@ swift_reflection_typeRefForMangledTypeName(SwiftReflectionContextRef ContextRef,
 ///
 /// The returned string is heap allocated and the caller must free() it when
 /// done.
-[[deprecated("Please use swift_reflection_copyNameForTypeRef()")]]
+__attribute__((deprecated("Please use swift_reflection_copyNameForTypeRef()")))
```

Comment:
This header may be included by MSVC as well right?  I think that we should use `LLVM_DEPRECATED`.

Reply:
That makes sense. I didn't realize we had that macro and didn't find it when looking for one.

LLVM defines it as:

```
#if defined(__clang__)
#define LLVM_DEPRECATED(MSG, FIX) __attribute__((deprecated(MSG, FIX)))
#else
#define LLVM_DEPRECATED(MSG, FIX) [[deprecated(MSG)]]
#endif
```

Is the `[[]]` syntax always going to be OK in the non-clang case? Seems like, if LLVM uses it, but I'm not sure if it uses this in C headers.

Reply:
I don't think that it is _always_ safe in the else case, but currently the CI is only covering clang and MSVC.  It should be okay with any C23 capable compiler.  I don't believe that these are guarded by `__cplusplus`.

Reply:
We don't include any LLVM headers here, so I stole the definition for a `SWIFT_REMOTE_MIRROR_DEPRECATED` that we can use here.